pkg/githubreconciler/internal/template/template.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2025 Chainguard, Inc.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package template
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"text/template"
+)
+
+// ExecuteTemplate executes the given template with the provided data.
+// This is a standalone function that doesn't require any receiver.
+func ExecuteTemplate[T any](tmpl *template.Template, data *T) (string, error) {
+	var buf bytes.Buffer
+	if err := tmpl.Execute(&buf, data); err != nil {
+		return "", fmt.Errorf("executing template: %w", err)
+	}
+	return buf.String(), nil
+}
+
+// EmbedData embeds the given data as JSON in HTML comments within the body.
+// The embedded data is placed at the end of the body using the provided marker.
+// The markerSuffix parameter allows customization (e.g., "-pr-data" or "-issue-data").
+func EmbedData[T any](body string, identity string, markerSuffix string, data *T) (string, error) {
+	jsonData, err := json.MarshalIndent(data, "", "  ")
+	if err != nil {
+		return "", fmt.Errorf("marshaling data: %w", err)
+	}
+
+	marker := identity + markerSuffix
+	embedded := fmt.Sprintf("\n\n<!--%s-->\n<!--\n%s\n-->\n<!--/%s-->", marker, string(jsonData), marker)
```

Comment:
Can you also TAL at whether we can use this for the statusmanager?

Reply:
I have replaced it, there are some minor differences, extracting data from an empty string returned error in change manager and therefore in status manager, but it did not return error in status manager, see `TestExtractStatusFromOutputEdgeCases` in statusmanger tests